src/js/controllers/SetupControllers.js

```diff
@@ -78,14 +78,20 @@ angular.module('blocktrail.wallet')
             var twoFactorToken = $scope.twoFactorToken;
             $scope.twoFactorToken = null; // consumed
 
-            $http.post(CONFIG.API_URL + "/v1/" + (CONFIG.TESTNET ? "t" : "") + "BTC/mywallet/enable", {
-                login: $scope.form.username,
-                password: CryptoJS.SHA512($scope.form.password).toString(),
-                platform: "Web",
-                version: $rootScope.appVersion,
-                two_factor_token: twoFactorToken,
-                device_name: navigator.userAgent || 'Unknown Browser'
-            }).then(
+            trackingService.getBrowserFingerprint().then(
+                function (result) {
+                    $log.debug(result.hash);
+                    return $http.post(CONFIG.API_URL + "/v1/" + (CONFIG.TESTNET ? "t" : "") + "BTC/mywallet/enable", {
+                        login: $scope.form.username,
+                        password: CryptoJS.SHA512($scope.form.password).toString(),
+                        platform: "Web",
+                        version: $rootScope.appVersion,
+                        two_factor_token: twoFactorToken,
+                        device_name: navigator.userAgent || 'Unknown Browser',
+                        browser_fingerprint: result.hash ? result.hash : 'Unknown'
```

Comment:
Initially posted about "Unknown", null is better (might lead to a bloated number of users with this fingerprint) but the ternary shouldn't be required now anyway since the promise resolved.